alacritty_terminal/src/tty/mod.rs

```diff
@@ -68,7 +68,7 @@ pub trait EventedReadWrite {
 #[derive(Debug, PartialEq, Eq)]
 pub enum ChildEvent {
     /// Indicates the child has exited.
-    Exited,
+    Exited(Option<i32>),
```

Comment:
Do we really need option instead of just saying that `0` is fine? Or maybe we should properly propagate an error from these calls?

Reply:
Yeah, this way is even better, I've switched over to `sucess: bool` model.

Reply:
Just to be clear here: `Option` does **not** come from `try_wait`, in that scenario the process hasn't quit. It is an `Option` because every process on Unix can either exit, or be killed by a signal. Processes killed by a signal have no exit status.